addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -394,111 +365,140 @@ public void handleCommand(ChannelUID channelUID, Command command) {
+                    } catch (ZoneMinderException ex) {
+                        logger.error(
+                                "{}: context='refreshThing' error in call to 'getHostCpuLoad' ExceptionClass='{}' - Message='{}'",
+                                getLogIdentifier(), ex.getClass().getCanonicalName(), ex.getMessage(), ex.getCause());
+                    }
 
-            if (hostLoad == null) {
-                logger.warn("{}: ZoneMinderHostLoad dataset could not be obtained (received 'null')",
-                        getLogIdentifier());
-            } else if (hostLoad.getHttpResponseCode() != 200) {
-                logger.warn(
-                        "BRIDGE [{}]: ZoneMinderHostLoad dataset could not be obtained (HTTP Response: Code='{}', Message='{}')",
-                        getThingId(), hostLoad.getHttpResponseCode(), hostLoad.getHttpResponseMessage());
+                    if (hostLoad == null) {
+                        logger.warn("{}: ZoneMinderHostLoad dataset could not be obtained (received 'null')",
+                                getLogIdentifier());
+                    } else if (hostLoad.getHttpStatus() != HttpStatus.OK_200) {
+                        logger.warn(
+                                "{}: ZoneMinderHostLoad dataset could not be obtained (HTTP Response: Code='{}', Message='{}')",
+                                getLogIdentifier(), hostLoad.getHttpStatus(), hostLoad.getHttpResponseMessage());
 
-            } else {
-                channelCpuLoad = hostLoad.getCpuLoad().toString();
-            }
+                    } else {
+                        channelCpuLoad = hostLoad.getCpuLoad().toString();
+                    }
 
-            if (fetchDiskUsage) {
-                IZoneMinderDiskUsage diskUsage = null;
-                try {
-                    diskUsage = zoneMinderServerProxy.getHostDiskUsage();
-                    logger.debug("{}: URL='{}' ResponseCode='{}' ResponseMessage='{}'", getLogIdentifier(),
-                            zoneMinderServerProxy.getHttpUrl(), zoneMinderServerProxy.getHttpResponseCode(),
-                            zoneMinderServerProxy.getHttpResponseMessage());
-                } catch (Exception ex) {
-                    logger.error("{}: Exception thrown in call to ZoneMinderDiskUsage: ", getLogIdentifier(), ex);
-                }
+                    if (curPriority.isPriorityActive(getBridgeConfig().getDiskUsageRefresh())) {
+                        IZoneMinderDiskUsage diskUsage = null;
+                        try {
+                            diskUsage = zoneMinderServerProxy.getHostDiskUsage();
+                            logger.debug("{}: URL='{}' ResponseCode='{}' ResponseMessage='{}'", getLogIdentifier(),
+                                    diskUsage.getHttpRequestUrl(), diskUsage.getHttpStatus(),
+                                    diskUsage.getHttpResponseMessage());
+                        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.